applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -483,6 +501,48 @@ void DistanceModificationProcess::SetDiscontinuousDistanceToSplitFlag()
     }
 }
 
+void DistanceModificationProcess::CheckAndStoreVariablesList(const std::vector<std::string>& rVariableStringArray) 
+{
+    const auto& r_node = *mrModelPart.NodesBegin(); 
+    if (rVariableStringArray.size()>0){
+        for (std::size_t i_variable=0; i_variable < rVariableStringArray.size(); i_variable++){
+            if (KratosComponents<Variable<double>>::Has(rVariableStringArray[i_variable])) {
+                const auto& r_double_var  = KratosComponents<Variable<double>>::Get(rVariableStringArray[i_variable]);
+                KRATOS_CHECK_DOF_IN_NODE(r_double_var, r_node);
+                KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(r_double_var, r_node)
+
+                mDoubleVariablesList.push_back(&r_double_var);               
+            }
+            else if (KratosComponents<ComponentType>::Has(rVariableStringArray[i_variable])){
+                const auto& r_component_var  = KratosComponents<ComponentType>::Get(rVariableStringArray[i_variable]);
+                KRATOS_CHECK_DOF_IN_NODE(r_component_var, r_node);
+                KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(r_component_var, r_node)
+               
+                mComponentVariablesList.push_back(&r_component_var);
+            }
+            else if (KratosComponents<Variable<array_1d<double,3>>>::Has(rVariableStringArray[i_variable])){
+                const auto& r_component_var_x = KratosComponents<ComponentType>::Get(rVariableStringArray[i_variable]+"_X");
+                const auto& r_component_var_y = KratosComponents<ComponentType>::Get(rVariableStringArray[i_variable]+"_Y");
+                const auto& r_component_var_z = KratosComponents<ComponentType>::Get(rVariableStringArray[i_variable]+"_Z");
+
+                KRATOS_CHECK_DOF_IN_NODE(r_component_var_x, r_node);
```

Comment:
To check `rVariableStringArray[i_variable]` must be enough. You don't need to go through each component.

Reply:
The variable in nodal data can be checked just once for the `array_1d<double,3>`, but DOF are added by component (also note that the z component will likely NOT be a dof in 3D)

Reply:
> (also note that the z component will likely NOT be a dof in 3D)

Did you mean in 2D? Would you add a condition to check if we are in 2D or 3D?

Reply:
Yes, I meant in 2D. I'm afraid we need to check for domain size (probably just get it from the modelpart?) and only use the Z component for 3D.

Reply:
> The variable in nodal data can be checked just once for the `array_1d<double,3>`, but DOF are added by component (also note that the z component will likely NOT be a dof in 3D)

I thought it was a nodal variable :man_facepalming:  forget about it.